core/src/main/scala/akka/persistence/jdbc/config/AkkaPersistenceConfig.scala

```diff
@@ -15,132 +15,132 @@ object ConfigKeys {
 }
 
 class SlickConfiguration(config: Config) {
-  val jndiName: Option[String] = config.as[String]("jndiName").trim
-  val jndiDbName: Option[String] = config.as[String]("jndiDbName")
+  val jndiName: Option[String] = config.asOptionalNonEmptyString("jndiName")
+  val jndiDbName: Option[String] = config.asStringOption("jndiDbName")
```

Comment:
I think the second one `jndiDbName` was missing the `.trim`. I don't see a reason to a have a empty string as a valid value for the `jndiDbName`. That said, we can have one single method that always return `None` for empty strings. 

And since it's internal API, we can keep its name short, eg: `asStringOption`